Module-11-Assignment/Question-1/EmailServiceForCreatingContacts.cls

```diff
@@ -0,0 +1,110 @@
+/**
+ * Email service class for creating contacts extracted from a XML file attached in the Mail.
+ */ 
+public class EmailServiceForCreatingContacts implements Messaging.InboundEmailHandler
+{
+    /**
+     * This method handles the every inbound emails and does the necessary operations.
+     */ 
+    public Messaging.InboundEmailResult handleInboundEmail(Messaging.InboundEmail email, Messaging.InboundEnvelope env)
+    {
+        Messaging.InboundEmailResult result = new Messaging.InboundEmailresult();
+        String contactsXmlFileContent = '';
+        if(email.textAttachments != null)
+        {
+            for(Messaging.InboundEmail.textAttachment attachment : email.textAttachments)
+            {
+                if(attachment.fileName == 'Contacts.xml')
+                {
+                    contactsXmlFileContent = attachment.Body;
+                    break;
+                }
+            }
+            if(String.isNotBlank(contactsXmlFileContent))
+            {
+                try
+                {
+                    parseAndCreateContacts(contactsXmlFileContent);
+                }
+                catch(Exception ex)
+                {
+                    result.success = false;
+                    result.message = ex.getMessage();
+                }
+            }
+            else
+            {
+                result.success = false;
+                result.message = 'Empty XML File was provided, So no data was inserted';
+            }
+        }
+        return result;
```

Comment:
send message for sucessful insert too indicating how many contacts were inserted

Reply:
Done